service/storage_service.cc

```diff
@@ -6880,6 +6878,18 @@ future<> storage_service::move_tablet(table_id table, dht::token token, locator:
             co_return;
         }
 
+        if (locator::contains(tinfo.replicas, dst.host)) {
+            throw std::runtime_error(format("Tablet {} has replica on {}", gid, dst.host));
+        }
+        auto src_dc_rack = get_dc_rack_for(host2ip(src.host)).value();
+        auto dst_dc_rack = get_dc_rack_for(host2ip(dst.host)).value();
```

Comment:
Don't we have a more modern API for this? In the form of the topology::node? (@bhalevy )

Reply:
Yes, we should take the info from token_metadata::topology